files/ansible/module_utils/test_result_parser/parser.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+import os
+import re
+
+from junitparser import TestCase, TestSuite, JUnitXml, Skipped, Error
+
+try:
+    from factory import TestFactory
+    from test_models import *
+except ImportError:
+    from ansible.module_utils.test_result_parser.factory import TestFactory
+    from ansible.module_utils.test_result_parser.test_models import *
+
+class TestParser(object):
+    def __init__(self, json_or_xml):
+        if str(json_or_xml) == "xml":
+            self.formatter = JUnitXml()
+            self.format = '.xml'
+        else:
+            # TODO: Implement a JSON format compatible with SQUAD
+            # This means using the python (l)xml ElementTree (<=> TestSuite())
+            self.formatter = None
+            self.format = '.json'
+
+        self.test_factory = TestFactory()
+
+    def main(self, results_path, test_threshold, output_path, extra=''):
+        result = ""
+        filename = os.path.basename(results_path)
+        with open(results_path, 'r') as output:
+            for line in output.readlines():
+                result += line
+
+        test, test_type = self.test_factory.getTest(result)
+        results, params = test.parse_output(result)
+
+        testcase = TestCase(test_type)
+
+        if test.failed(results, test_threshold):
+            testcase.result = Error(test.fail_msg.format(test_threshold), test_type)
+
+        suite = TestSuite(test_type + extra)
+        for name, value in results.items():
```

Comment:
if you're going to bundle results and params (whatever they mean) into one block, then what's the point of separating them?

Reply:
Right now they are put into the default junitparser XML object 'property'.
But junitparser allows for defining custom 'Properties'.
The only problem is that there is no infrastructure yet that can display the complete output.
So waiting for that I'm parsing params (environment properties) and results separately and dumping them directly in the JUnit/XUnit.

Reply:
Makes sense, let's leave this for a later decision.